serialization/homework.py

```diff
@@ -16,4 +16,131 @@
 Advanced
 Добавити опрацьовку формату ini
 
-"""
\ No newline at end of file
+"""
+import json
+import pickle
+import uuid
+
+from ruamel import yaml
+from ruamel.yaml import YAML
+
+from objects_and_classes.homework.homework import Cesar, Garage, Car
+
+
+# yaml
+class SerCesar(Cesar):
+    yaml = YAML()
+
+    def dict(self):
+        data = {'name': self.name, 'register_id': str(self.register_id),
+                'garages': [garage.dict() for garage in self.garages] if self.garages else [], }
+
+        return data
+
+    @staticmethod
+    def from_yaml_string(data):
+        return SerCesar.from_yaml(yaml.load(data))
+
+    def to_yaml(self):
+        return yaml.dump(self.dict())
+
+    def to_yaml_file(self):
+        with open("cesar_to_yaml.yaml", 'w') as file:
+            self.yaml.dump(self.dict(), file)
+
+    @classmethod
+    def from_yaml(cls, data):
+        name = data['name']
+        garages = [SerGarage.from_yaml_string(garage) for garage in data['garages']]
+        return SerCesar(name=name, garages=garages)
+
+    def instance_from_yaml_string(obj, yaml_string: str):
+        return obj.from_yaml(yaml.load(yaml_string))
+
+    @staticmethod
+    def from_yaml_file():
+        with open("cesar_to_yaml.yaml", 'r') as file:
+            return SerCesar.from_yaml(yaml.load(file))
+
+
+# json
+class SerCar(Car):
+
+    def dict(self):
+        return {'price': self.price, 'car_type': self.car_type, 'producer': self.producer, 'mileage': self.mileage,
+                'number': str(self.number)}
+
+    def to_json(self):
+        return json.dumps(self.dict())
+
+    def to_json_file(self):
+        with open("car_to_file.json", 'w') as file:
+            json.dump(self.dict(), file)
+
+    @classmethod
+    def from_json(cls, data):
+        if not isinstance(data, dict):
+            data = json.loads(data)
+
+        number = data['number']
+        price = data['price']
+        car_type = data['car_type']
+        producer = data['producer']
+        mileage = data['mileage']
+
+        return cls(price=price, car_type=car_type, producer=producer, mileage=mileage, number=number)
+
+    @classmethod
+    def from_json_file(cls):
+        with open("car_to_file.json", 'r') as file:
```

Comment:
`file` - зарезервоване слово для позначення типу даних. Старайся не називати так змінні
```suggestion
        with open("car_to_file.json", 'r') as f:
```